test/LndClient/RPCSpec.hs

```diff
@@ -305,6 +305,26 @@ spec = do
                             Left {} -> False
                             Right xs -> any ((cp ==) . CloseChannel.chPoint) xs
                         )
+  it "subscriptionCanBeCancelled" $ withEnv $ do
+    bob <- getLndEnv Bob
+    alice <- getLndEnv Alice
+    GetInfoResponse bobPubKey _ _ <- liftLndResult =<< getInfo bob
+    let openChannelRequest =
+          OpenChannelRequest
+            { nodePubkey = bobPubKey,
+              localFundingAmount = MSat 20000000,
+              pushSat = Just $ MSat 1000000,
+              targetConf = Nothing,
+              satPerByte = Nothing,
+              private = Nothing,
+              minHtlcMsat = Nothing,
+              remoteCsvDelay = Nothing,
+              minConfs = Nothing,
+              spendUnconfirmed = Nothing,
+              closeAddress = Nothing
+            }
+    a <- spawnLink $ liftLndResult =<< openChannel (const $ return ()) alice openChannelRequest
+    liftIO $ cancel a
```

Comment:
Am I right that Async cancelling (or race, which is the same) was not working in proto3-suite?
Now it works fine.

Reply:
Yes, race was not working properly, because in many cases `cancel` is a part of `race` flow. If this is working now - it's really good. But test without assert looks strange. I suggest to test not just "some" cancellation, but fact that it's done immediately. For example instead of `liftIO $ cancel a` we can do something like 

```haskell
liftIO $ do
  res <- race (delay 100000) $ cancel a
  res `shouldSatisfy` isRight
```

Reply:
Agree